client/daemon.go

```diff
@@ -242,11 +246,14 @@ func (d daemon) processKey(keyID string) error {
 	if err != nil {
 		return fmt.Errorf("Error marshalling key %s: %s", keyID, err.Error())
 	}
-	err = ioutil.WriteFile(d.keyFilename(keyID), b, defaultFilePermission)
+	err = ioutil.WriteFile(d.tmpFilename(keyID), b, defaultFilePermission)
 	if err != nil {
 		return fmt.Errorf("Error writing key %s to file: %s", keyID, err.Error())
 	}
-
+	err = os.Rename(d.tmpFilename(keyID), d.keyFilename(keyID))
+	if err != nil {
+		return fmt.Errorf("Error renaming key %s temporary file: %s", keyID, err.Error())
```

Comment:
If this fails, we might end up in a state where we have lingering .key.tmp files. it may be better to add in a `defer os.Remove(tmpFilename(id))` to clean up if the writefile or rename call fails.

Reply:
Won't defer since we don't always want to rm (i.e. if we succeed in renaming)